service/lambda/api_op_AddLayerVersionPermission.go

```diff
@@ -208,6 +210,54 @@ func (m *opAddLayerVersionPermissionResolveEndpointMiddleware) HandleSerialize(c
 		)
 	}
 
+	authSchemes, err := internalauth.GetAuthenticationSchemes(&resolvedEndpoint.Properties)
+	if err != nil {
+		var nfe *internalauth.NoAuthenticationSchemesFoundError
+		if errors.As(err, &nfe) {
```

Comment:
**correctness**: What if the error _isn't_ `NoAuthenticationSchemesFoundError`? It looks like we're just dropping the error.